### PR TITLE
Add support for ignoring file patterns

### PR DESCRIPTION
**Pull request by mmeyer724**

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * Borrowed from actions/cache on Github
+ * https://github.com/actions/cache/blob/eed9cfe64d00dd64cbb36bec915d41dd6fab9f6c/src/utils/actionUtils.ts#L49
+ */
+export function getInputAsArray(
+  name: string,
+  options?: core.InputOptions
+): string[] {
+  return core
+    .getInput(name, options)
+    .split("\n")
```
Comment (myfashionhub) [inline src/utils.ts:13]: single quotes

File: src/main.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
-import * as path from 'path';
-import { existsSync } from 'fs';
 import * as core from '@actions/core';
 import * as github from '@actions/github';
+import { existsSync } from 'fs';
+import * as path from 'path';
+import picomatch from 'picomatch';
```
Comment (myfashionhub) [inline src/main.ts:5]: don't commit `node_modules/picomatch/lib/.DS_Store`

Comment (mmeyer724) [reply src/main.ts:5]: Ahhh great catch on that, I still need to add this to my global gitignore! Thanks

Comment (mmeyer724) [reply src/main.ts:5]: Screen Shot 2020-08-26 at 10 21 22 AM
Hmm looks like it's actually being removed, not added 😅 I think picomatch (the library itself) included it in a previous version, and then they must've removed it in the newer version I'm installing.

Either way, this was nice reminder that I needed to update my own global gitignore

**Comment by mmeyer724** [comment]
@myfashionhub Thanks, I squashed all of the commits into one

File: src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import * as core from '@actions/core';
+
+/*
+ * Borrowed from actions/cache on Github
+ * https://github.com/actions/cache/blob/eed9cfe64d00dd64cbb36bec915d41dd6fab9f6c/src/utils/actionUtils.ts#L49
+ */
+export function getInputAsArray(
+  name: string,
+  options?: core.InputOptions
+): string[] {
```
Comment (maecapozzi) [inline src/utils.ts:10]: Is it necessary to define the return type here, or can it just be inferred by the compiler?

File: lib/utils.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * https://github.com/actions/cache/blob/eed9cfe64d00dd64cbb36bec915d41dd6fab9f6c/src/utils/actionUtils.ts#L49
+ */
+function getInputAsArray(name, options) {
+    return core
+        .getInput(name, options)
+        .split('\n')
+        .map(s => s.trim())
+        .filter(x => x !== '');
+}
+exports.getInputAsArray = getInputAsArray;
```
Comment (mmeyer724) [inline lib/utils.js:22]: note: everything under `lib/` is compiled with tsc, this isn't my actual code. 🙂 The .ts files are lower down in this PR

Unfortunately this GitHub Action requires all artifacts (including node_modules and compiled JS) be pushed to the repo.

File: src/main.ts [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
 import { CHECK_NAME, EXTENSIONS_TO_LINT } from './constants';
 import { eslint } from './eslint-cli';
+import * as utils from './utils';
```
Comment (maecapozzi) [inline src/main.ts:8]: Why not just import only `getInputAsArray`? What's the benefit of importing `*` when you only need one function?

File: src/utils.ts (hunk 3 above)
Comment (mmeyer724) [reply src/utils.ts:10]: probably not necessary, but I copied this function from https://github.com/actions/cache/blob/eed9cfe64d00dd64cbb36bec915d41dd6fab9f6c/src/utils/actionUtils.ts#L49 so I would rather leave it unchanged unless there's a strong reason to change it

**Pull request merged**